pkg/util/kubernetes/autoscalers/processor_test.go

```diff
@@ -173,6 +175,117 @@ func TestProcessor_UpdateExternalMetrics(t *testing.T) {
+		in         map[string]custommetrics.ExternalMetricValue
+		out        []datadog.Series
+		l          sync.Mutex
+		batchCalls int
+		err        error
+		timeout    bool
+	}{
+		{
+			desc: "one batch",
+			in: lambdaMakeChunks(14, custommetrics.ExternalMetricValue{
+				MetricName: "foo",
+				Labels:     map[string]string{"foo": "bar"}}),
+			out: []datadog.Series{
+				{
+					Metric: &metricName,
+					Points: []datadog.DataPoint{
+						makePoints(1531492452000, 12),
+						makePoints(penTime, 14), // Force the penultimate point to be considered fresh at all time(< externalMaxAge)
+						makePoints(0, 27),
+					},
+					Scope: makePtr("foo:bar"),
+				},
+			},
+			batchCalls: 1,
+			err:        nil,
+			timeout:    false,
+		},
+		{
+			desc: "several batches",
+			in: lambdaMakeChunks(158, custommetrics.ExternalMetricValue{
+				MetricName: "foo",
+				Labels:     map[string]string{"foo": "bar"}}),
+			out: []datadog.Series{
+				{
+					Metric: &metricName,
+					Points: []datadog.DataPoint{
+						makePoints(1531492452000, 12),
+						makePoints(penTime, 14), // Force the penultimate point to be considered fresh at all time(< externalMaxAge)
+						makePoints(0, 27),
+					},
+					Scope: makePtr("foo:bar"),
+				},
+			},
+			batchCalls: 4,
+			err:        nil,
+			timeout:    false,
+		},
+		{
+			desc: "several batches, one error",
+			in: lambdaMakeChunks(158, custommetrics.ExternalMetricValue{
+				MetricName: "foo",
+				Labels:     map[string]string{"foo": "bar"}}),
+			out: []datadog.Series{
+				{
+					Metric: &metricName,
+					Points: []datadog.DataPoint{
+						makePoints(1531492452000, 12),
+						makePoints(penTime, 14), // Force the penultimate point to be considered fresh at all time(< externalMaxAge)
+						makePoints(0, 27),
+					},
+					Scope: makePtr("foo:bar"),
+				},
+			},
+			batchCalls: 4,
+			err:        fmt.Errorf("Networking Error, timeout!!!"),
+			timeout:    true,
+		},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprintf("#%d %s", i, tt.desc), func(t *testing.T) {
+			datadogClient := &fakeDatadogClient{
+				queryMetricsFunc: func(int64, int64, string) ([]datadog.Series, error) {
+					tt.l.Lock()
+					defer tt.l.Unlock()
+					tt.batchCalls -= 1
+					if tt.timeout == true && tt.batchCalls == 1 {
+						// Error will be under the format:
+						// Error: Error while executing metric query avg:foo-56{foo:bar}.rollup(30),avg:foo-93{foo:bar}.rollup(30),[...],avg:foo-64{foo:bar}.rollup(30),avg:foo-81{foo:bar}.rollup(30): Networking Error, timeout!!!
+						// In the logs, we will be able to see which bundle failed, but for the tests, we can't know which routine will finish first (and therefore have `bc == 1`), so we only check the error returned by the Datadog Servers.
+						return nil, fmt.Errorf("Networking Error, timeout!!!")
+					}
+					return tt.out, nil
+				},
+			}
+			p := &Processor{datadogClient: datadogClient}
+
+			_, err := p.validateExternalMetric(tt.in)
+			if err != nil {
```

Comment:
I could be mistaken but if `err` is `nil` but `tt.err` is not `nil` that won't get asserted here, correct?

Reply:
correct indeed